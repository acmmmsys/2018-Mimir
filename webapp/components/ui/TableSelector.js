import React, { Component } from 'react';
import { Icon, Label, Menu, Table } from 'semantic-ui-react'
import classnames from 'classnames';

import 'style/ui/TableSelector.scss'

const defaultClassNames = [
    'selection-table',
    'analysis-table',
];

class TableSelector extends Component {

    static defaultProps = {
        classNames: [],
        header: [],
        content: [],
    }

    buildHeaderRows = () => {
        return (
            <Table.Row className='table-selection-row'> 
                {this.props.header.map(header => (
                    <Table.HeaderCell className='table-selection-header-cell'>
                        {header}
                    </Table.HeaderCell>
                ))}
            </Table.Row>
        );
    }

    buildBodyRows = () => {
        return this.props.content.map((content, i) => {
            
            let classNames = classnames({
                'table-selection-row': true,
                'clickable': true,
                'selected': i === this.props.selectedRow,
            });

            return (
                <Table.Row onClick={() => this.props.onClick(i)} className={classNames}>
                    {content.map(element => (
                        <Table.Cell>
                            {element}
                        </Table.Cell>
                    ))}
                </Table.Row>
            )
        });
    }

    render() {

        let classNames = classnames(
            defaultClassNames,
            this.props.classNames,
        );
        
        return (
            <div className={"test"}>
                <Table basic className={classNames}>
                    <Table.Header>
                        {this.buildHeaderRows()}
                    </Table.Header>
                    <Table.Body className='selection-table-body'>
                        {this.buildBodyRows()}
                    </Table.Body>
                </Table>
            </div>
        );
    }
}

export default TableSelector