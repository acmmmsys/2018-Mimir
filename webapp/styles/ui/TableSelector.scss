@import '../colors';

.selected {
    background-color: darken( $backgroundColor, 5% )
}

.selection-table {
    color: $defaultTextColor!important;
    overflow-y: scroll;
}

.table-selection-row {
    display: flex;
    color: $defaultTextColor!important;
    align-content: space-between;
    justify-content: space-between;
    
    &:hover {
        background-color: lighten( $backgroundColor, 5% )
    }
}

.table-selection-header {
    display: flex;
    align-content: space-between;
}

.selection-table-body {
    min-height: 100%;
    overflow-y: scroll;
}

.table-selection-header-cell {
    color: $defaultTextColor!important;
}

.analysis-table {
    align-self: flex-start;
    color: $defaultTextColor!important;
}