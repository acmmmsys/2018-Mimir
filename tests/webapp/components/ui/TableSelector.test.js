import React from 'react';
import TableSelector from '../../../../webapp/components/ui/TableSelector'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
 
const mockStore = configureStore();

let store;
let wrapper;

test('TableSelector does render', () => {
  const wrapper = shallow(
      <TableSelector store={mockStore(initialState)}>This is a test!</TableSelector>
  );
  expect(wrapper).toMatchSnapshot();
});
