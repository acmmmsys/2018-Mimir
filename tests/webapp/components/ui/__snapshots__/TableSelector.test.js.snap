// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableSelector does render 1`] = `
<div
  className="test"
>
  <Table
    as="table"
    basic={true}
    className="selection-table analysis-table "
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
        className="table-selection-row"
      />
    </TableHeader>
    <TableBody
      as="tbody"
      className="selection-table-body"
    />
  </Table>
</div>
`;
